File: Data Structures & Algorithms/is-palindrome/submission-1.py
class Solution:
    def isPalindrome(self, s: str) -> bool: 
        '''
        Time Complexity: O(n)
        Space Complexity: O(n) n = len(s)
        '''
        ans = ''
        for c in s:
            if c.isalnum():
                ans += c.lower()
        return ans == ans[::-1]

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        /*
        itterate through nums
        push to heap
        while top of heap idx is outside of window -> pop
        push top of heap to ans vector

        Time Complexity: O(nlogn) Space Complexity: O(n)
        */
        vector<int> ans;
        priority_queue<pair<int, int>> maxHeap; // {val, idx}
        for (int i{0}; i < nums.size(); i++) {
            maxHeap.push({nums[i], i});
            while (maxHeap.top().second < i - k + 1) maxHeap.pop();
            if (i >= k - 1) ans.push_back(maxHeap.top().first);
        }
        return ans;
    }
};
